gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -564,6 +587,14 @@ public boolean shouldAutoClose() {
     return true;
   }
 
+  /**
+   * @return list of provided headers that will be sent with GRPC call with any duplicates removed
+   *     see {@link #removeCredentialDuplicateHeaders()}
+   */
+  public Map<String, String> getHeadersWithDuplicatesRemoved() {
```

Comment:
Is this only used in tests? If it is, we can make this package private and mark it as `@VisibleForTesting`? Or we don't have to expose another method, just make `removeCredentialDuplicateHeaders` package privare.